### ISSUE TITLE
Geomeppy and wwr function problem 

### ISSUE THREAD

**Issue by orcunkoraliseri** [santoshphilip/eppy#174]
While setting WWR with geomeppy, i faced the problem of 'Called Process Error'. Can you propose me a solution to close subprocess of wwr ?

When i checked the idf file with set_wwr() function, it changed the structure of idf.

```
from geomeppy import IDF 
from eppy.results import readhtml
import random as rd

IDF.setiddname("C:/EnergyPlusV8-8-0/Energy+.idd") 
idfname = "C:/Users/orcun/Desktop/eppy/E_eppy01/OpenStudio/E_eppy01/ModelToIdf/in.idf"
epwfile="C:/Users/orcun/Desktop/eppy/TUR_Izmir_Izmir_TurTMY.epw"
idf = IDF(idfname, epwfile)

WWR_list=[0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9] # set window to wall ratio external walls
x2 = rd.choice(WWR_list)
idf.set_wwr(x2)

#idf.printidf()
#idf.view_model()

idf.run()
```

**Comment by jamiebull1** [comment]
@orcunkoraliseri Can you post the full traceback please?

**Comment by orcunkoraliseri** [comment]
CalledProcessError                        Traceback (most recent call last)
<ipython-input-1-ad86d9bbcde8> in <module>()
     78 #idf.set_default_constructions()
     79 #idf.printidf()
---> 80 idf.run()
     81 
     82 '''

C:\Users\orcun\Anaconda2\lib\site-packages\eppy\modeleditor.pyc in run(self, **kwargs)
   1002         self.saveas('in.idf')
   1003         # run EnergyPlus
-> 1004         run(self, self.epw, **kwargs)
   1005         # remove in.idf
   1006         os.remove('in.idf')

C:\Users\orcun\Anaconda2\lib\site-packages\eppy\runner\run_functions.pyc in run(idf, weather, output_directory, annual, design_day, idd, epmacro, expandobjects, readvars, output_prefix, output_suffix, version, verbose, ep_version)
    236     try:
    237         if verbose == 'v':
--> 238             check_call(cmd)
    239         elif verbose == 'q':
    240             check_call(cmd, stdout=open(os.devnull, 'w'))

C:\Users\orcun\Anaconda2\lib\subprocess.pyc in check_call(*popenargs, **kwargs)
    539         if cmd is None:
    540             cmd = popenargs[0]
--> 541         raise CalledProcessError(retcode, cmd)
    542     return 0
    543

**Comment by jamiebull1** [comment]
Thanks. It looks like EnergyPlus is having trouble running the IDF. Is there a `.err` file produced?

Also, in future can you raise issues with Geomeppy on the [geomeppy github](https://github.com/jamiebull1/geomeppy)? I'm sure @santoshphilip doesn't need to be bothered by them.

**Comment by orcunkoraliseri** [comment]
You are right. Sorry about it.

**Comment by jamiebull1** [comment]
No problem. I can handle this one here.

**Comment by orcunkoraliseri** [comment]
This is the produced .err file. 

Program Version,EnergyPlus, Version 8.8.0-7c3bbe4830, YMD=2017.12.19 16:34,IDD_Version 8.8.0
   ************* Beginning Zone Sizing Calculations
   ** Warning ** ProcessScheduleInput: Schedule:Constant="ALWAYS ON DISCRETE", Blank Schedule Type Limits Name input -- will not be validated.
   ** Warning ** ProcessScheduleInput: Schedule:Constant="ALWAYS OFF DISCRETE", Blank Schedule Type Limits Name input -- will not be validated.
   ** Warning ** ProcessScheduleInput: Schedule:Constant="ALWAYS ON CONTINUOUS", Blank Schedule Type Limits Name input -- will not be validated.
   ** Severe  ** Did not find matching material for Construction EXTERIOR WINDOW, missing material = ECREF-2 COLORED 6MM
   ** Severe  ** Did not find matching material for Construction EXTERIOR WINDOW, missing material = AIR 13MM
   ** Severe  ** Did not find matching material for Construction EXTERIOR WINDOW, missing material = ECREF-2 COLORED 6MM
   ** Severe  ** Nominal U is zero, for construction=EXTERIOR WINDOW
   ** Warning ** GetHTSurfaceData: Surfaces with interface to Ground found but no "Ground Temperatures" were input.
   **   ~~~   ** Found first in surface=A_SRF_4
   **   ~~~   ** Defaults, constant throughout the year of (18.0) will be used.
   **  Fatal  ** GetSurfaceData: Errors discovered, program terminates.
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=4
   ..... Last severe error=Nominal U is zero, for construction=EXTERIOR WINDOW
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 4 Warning; 4 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 4 Warning; 4 Severe Errors; Elapsed Time=00hr 00min  0.89sec

**Comment by jamiebull1** [comment]
```
** Severe ** Did not find matching material for Construction EXTERIOR WINDOW, missing material = ECREF-2 COLORED 6MM
** Severe ** Did not find matching material for Construction EXTERIOR WINDOW, missing material = AIR 13MM
** Severe ** Did not find matching material for Construction EXTERIOR WINDOW, missing material = ECREF-2 COLORED 6MM
```

Were these materials in your IDF before you set the WWR?

**Comment by orcunkoraliseri** [comment]
Actually, this is the part i did not understand because when i print window material and window construction, i am facing with this output. On the contrary, the .err file extract 'ECREF-2 COLORED 6MM' that material as a window material.


     Construction,
        Exterior Window,          !- Name
        Clear 3mm,                !- Outside Layer
        Air 13mm,                 !- Layer 2
        Clear 3mm;                !- Layer 3


     WindowMaterial:Glazing,
        Clear 3mm,                !- Name
        SpectralAverage,          !- Optical Data Type
        ,                         !- Window Glass Spectral Data Set Name
        0.003,                    !- Thickness
        0.837,                    !- Solar Transmittance at Normal Incidence
        0.075,                    !- Front Side Solar Reflectance at Normal Incidence
        0.0,                      !- Back Side Solar Reflectance at Normal Incidence
        0.898,                    !- Visible Transmittance at Normal Incidence
        0.081,                    !- Front Side Visible Reflectance at Normal Incidence
        0.0,                      !- Back Side Visible Reflectance at Normal Incidence
        0.0,                      !- Infrared Transmittance at Normal Incidence
        0.84,                     !- Front Side Infrared Hemispherical Emissivity
        0.84,                     !- Back Side Infrared Hemispherical Emissivity
        0.9,                      !- Conductivity
        1.0;                      !- Dirt Correction Factor for Solar and Visible Transmittance

**Comment by jamiebull1** [comment]
Do you have any windows in the model before running `IDF.set_wwr()`? If so could you post an example, and also of the ones created by Geomeppy.

**Comment by orcunkoraliseri** [comment]
I am currently new for coding and energy plus because of that i used grasshopper/honeybee to produce idf file but when i checked the idf file that honeybee gave it to me there is 'Clear 3mm' material as a glazing. Also i worked eppy with this idf file. 

Besides, i searched the material of 'ECREF-2 COLORED 6MM' as a glazing material, i found that in energyplus window materials of idf file. But i did not use that file before.

**Comment by jamiebull1** [comment]
My guess is that grasshopper is creating windows using `ECREF-2 COLORED 6MM` but I can't tell this for sure unless you post the examples.

The other thing I would like to know from the examples I asked for is what type of window object is in the IDF. Geomeppy currently assumes `FENESTRATIONSURFACE:DETAILED` which might not be the case.

**Comment by jamiebull1** [comment]
@orcunkoraliseri did you manage to resolve your issue?

**Comment by orcunkoraliseri** [comment]
Thank you for asking. Sadly, i did not. As you said i removed all the fenestration surfaces before applying wwr and keeping the same glazing material but still it did not work maybe because of grasshopper/honeybee tool. Therefore, i am currently working on to create an idf file from zero with geomeppy and eppy. I think it is better that way.

In addition to that, i decided to start learn structure of idf file. If it is possible could you offer me some sources related with that?

**Comment by jamiebull1** [comment]
I'm interested in what's going wrong at the geomeppy side. Would you be able to share the IDF that you were passing in as `C:/Users/orcun/Desktop/eppy/E_eppy01/OpenStudio/E_eppy01/ModelToIdf/in.idf`?

As for IDF structure, that's a big question. I'd recommend looking at the example files in your EnergyPlus install directory. There's a lot of good stuff in there that you should be able to use to guide your building with eppy/geomeppy.

**Comment by orcunkoraliseri** [comment]
As you said i changed the idf file with energyplus directory. there is another problem has occured. I updated the geomeppy to 0.3.9 but `from geomeppy import IDF` does not work.

**Comment by orcunkoraliseri** [comment]
By the way thank you for close support. I am sharing with you the files that i extracted from grashopper/honeybee.  The second one contains glazing and first one does not. File names are different from the one you mentioned but they are the same idf file.

[E_eppy03_noWWR.zip](https://github.com/santoshphilip/eppy/files/1581787/E_eppy03_noWWR.zip)
[E_eppy03_WWR.zip](https://github.com/santoshphilip/eppy/files/1581788/E_eppy03_WWR.zip)

**Comment by jamiebull1** [comment]
@orcunkoraliseri This should be resolved as of geomeppy `0.3.12`. Thanks for your help.
